keras/engine/training.py

```diff
@@ -962,10 +962,21 @@ def _make_predict_function(self):
                                                name='predict_function',
                                                **kwargs)
 
+    def _check_num_samples(self, ins, batch_size=None, steps=None, steps_name='steps'):
+        if steps is not None:
+            num_samples = steps
```

Comment:
That should be, on average `step * batch_size`. However, since the data length may not be a multiple of `batch_size`, and because in the generator case there may be batches of different sizes, it is actually impossible to determine `num_samples`.